src/index/ivf/ivfrbq_wrapper.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2019-2025 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "index/ivf/ivfrbq_wrapper.h"
+
+#include <faiss/cppcontrib/knowhere/impl/CountSizeIOWriter.h>
+
+#include <stdexcept>
+
+#include "faiss/IndexPreTransform.h"
+#include "faiss/index_io.h"
+
+namespace knowhere {
+
+IndexIVFRaBitQWrapper::IndexIVFRaBitQWrapper(std::unique_ptr<faiss::Index>&& index_in)
+    : Index{index_in->d, index_in->metric_type}, index{std::move(index_in)} {
+    ntotal = index->ntotal;
+    is_trained = index->is_trained;
+    verbose = index->verbose;
+    metric_arg = index->metric_arg;
+}
+
+IndexIVFRaBitQWrapper::~IndexIVFRaBitQWrapper() {
+}
+
+void
+IndexIVFRaBitQWrapper::train(faiss::idx_t n, const float* x) {
+    // index->train(n, x);
```

Comment:
Prefer this form over throwing exceptions; it seems that IndexIVFRaBitQWrapper has training logic, but it's written in ivf.cc instead.